crates/hir/src/lib.rs

```diff
@@ -1418,7 +1418,8 @@ impl Adt {
     }
 
     pub fn layout(self, db: &dyn HirDatabase) -> Result<Layout, LayoutError> {
-        if db.generic_params(self.into()).iter().count() != 0 {
+        let generic_params = &db.generic_params(self.into());
+        if generic_params.iter().count() != 0 || generic_params.iter_lt().count() != 0 {
```

Comment:
Nit: `.iter().next().is_some()`

Reply:
Yeah, that could be better, since count consumes the entire iterator.